src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -3692,6 +3697,1485 @@ explicit_encryption_destroy (ee_fixture *eef)
+                                   "}",
+                                   "}",
+                                   "]");
+      mongoc_cursor_t *cursor =
+         test_explicit_encryption_range_agg_helper (eef_range, eef);
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_MATCH (got, "{ 'encryptedInt': 6}");
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_MATCH (got, "{ 'encryptedInt': 30}");
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_MATCH (got, "{ 'encryptedInt': 0}");
+      ASSERT (!mongoc_cursor_next (cursor, &got) &&
+              "expected three documents, got more than three");
+   }
+   // aggregate command to return one document
+   {
+      const bson_t *got;
+      eef_range->query = BCON_NEW ("$and",
+                                   "[",
+                                   "{",
+                                   "encryptedInt",
+                                   "{",
+                                   "$gte",
+                                   BCON_INT32 (7),
+                                   "}",
+                                   "}",
+                                   "{",
+                                   "encryptedInt",
+                                   "{",
+                                   "$lte",
+                                   BCON_INT32 (30),
+                                   "}",
+                                   "}",
+                                   "]");
+      mongoc_cursor_t *cursor =
+         test_explicit_encryption_range_agg_helper (eef_range, eef);
+      ASSERT (mongoc_cursor_next (cursor, &got));
+      ASSERT_MATCH (got, "{ 'encryptedInt': 30}");
+      ASSERT (!mongoc_cursor_next (cursor, &got) &&
+              "expected one document, got more than one");
+   }
+   // aggregate command to return zero documents
+   // The difference between this and the previous test is using lt and not lte
+
+   {
+      const bson_t *got;
+      eef_range->query = BCON_NEW ("$and",
+                                   "[",
+                                   "{",
+                                   "encryptedInt",
+                                   "{",
+                                   "$gte",
+                                   BCON_INT32 (7),
+                                   "}",
+                                   "}",
+                                   "{",
+                                   "encryptedInt",
+                                   "{",
+                                   "$lt",
+                                   BCON_INT32 (30),
+                                   "}",
+                                   "}",
+                                   "]");
+      mongoc_cursor_t *cursor =
+         test_explicit_encryption_range_agg_helper (eef_range, eef);
+      ASSERT (!mongoc_cursor_next (cursor, &got) &&
+              "expected zero documents, got more than none");
+   }
```

Comment:
Suggest adding a test case to test explicit encryption with an Aggregate Expression.

An Aggregate Expression can be passed to a find command with `$expr`. Here is an example in the mongo shell:

```
> db.coll.find({$expr: { $and: [ { $gt: [ "$encryptedInt", 0 ] } ] }})
```